### PR TITLE
4 packages from sneeuwballen/zipperposition at 2.1

### PR DESCRIPTION
**Pull request by c-cube**
This pull-request concerns:
-`libzipperposition.2.1`: Library for Zipperposition
-`logtk.2.1`: Core types and algorithms for logic
-`zipperposition.2.1`: A fully automatic theorem prover for typed higher-order and beyond
-`zipperposition-tools.2.1`: Support tools for Zipperposition

---
* Homepage: https://github.com/sneeuwballen/zipperposition
* Source repo: git+https://github.com/sneeuwballen/zipperposition.git
* Bug tracker: https://github.com/sneeuwballen/zipperposition/issues

---
:camel: Pull-request generated by opam-publish v2.1.0

### CODE REVIEW

**Comment by c-cube** [comment]
seems like the lower bound is pretty benign

**Comment by c-cube** [comment]
I was hoping the lower bound thing would do some kind of bisect to find the highest failing lower bound, but it seems not :(

**Comment by avsm** [comment]
suggestions for algorithms to constraint solve that lower-bound bisection are welcome :-)

File: packages/logtk/logtk.2.1/opam [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  "qcheck-alcotest" {with-test & >= "0.9"}
+  "ocaml" {>= "4.07"}
+]
+depopts: [
+  "msat"
+]
+conflicts: [
+  "msat" { < "0.8.1" }
+  "msat" { >= "0.10" }
+  "zarith" { < "1.4" }
```
Comment (mseri) [inline packages/logtk/logtk.2.1/opam:33]:
```suggestion
   "zarith" { < "1.7" }
```

**Comment by c-cube** [comment]
> suggestions for algorithms to constraint solve that lower-bound bisection are welcome :-)

I'm a bit confused because my naive guess at how the lower bound thing works, is that it would try lower and lower bounds. But I imagine it "just" adds a constraint to use as low bounds as possible, which means only one build.

Well, technically it could mean several builds as you can have several uncomparable solutions which minimize the version of different libraries. But in any case, bisecting would mean walking the lattice of solutions that are greater than these minima, and trying to build on them, which can be too costly even for a build cluster, so I'm not claiming I have a practical solution :slightly_smiling_face:

**Comment by mseri** [comment]
Almost ready. Are the two test failures spurious?

can you add the missing licenses please?

    2021-09-03 15:20.21: Error in zipperposition-tools.2.1: The field 'license' is not present.
    2021-09-03 15:20.21: Error in zipperposition.2.1: The field 'license' is not present.
    2021-09-03 15:20.21: Error in logtk.2.1: The field 'license' is not present.
    2021-09-03 15:20.21: Error in libzipperposition.2.1: The field 'license' is not present.

**Comment by mseri** [comment]
Thanks!

**Comment by kit-ty-kate** [comment]
Thanks! Sorry for the delay

**Pull request merged**